reinforcement/algorithm/reinforce.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+
+
+def _calc_signal(ops, action_dist, returns):
+    return ops.reduce_mean(ops.log(action_dist) * returns)
```

Comment:
This will be tackled once I've implemented general reinforcement learning tests. These should detect this issue reliably.